servicedirectory/src/main/java/com/example/snippets/ServicesResolve.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.servicedirectory.snippets;
+
+// [START servicedirectory_resolve_service]
+
+import com.google.cloud.servicedirectory.v1beta1.Endpoint;
+import com.google.cloud.servicedirectory.v1beta1.LookupServiceClient;
+import com.google.cloud.servicedirectory.v1beta1.ResolveServiceRequest;
+import com.google.cloud.servicedirectory.v1beta1.ResolveServiceResponse;
+import com.google.cloud.servicedirectory.v1beta1.ServiceName;
+import java.io.IOException;
+
+public class ServicesResolve {
+
+  public static void resolveService() throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
```

Comment:
Not sure this is the ideal pattern for setting values.

Reply:
This is from the SAMPLE_FORMAT.md  https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#function-structure

Reply:
Yep, this is good.